[Merge] Merge tag 'jdk8u282-ga'(jdk)

yuleil opened the pull request:
Summary: Merge 'jdk8u282-ga' from OpenJDK

Test Plan: CI pipeline

Reviewed-by: joeyleeeeeee97, sanhong

Issue: alibaba/dragonwell8#193

Commit:
8008657: JSpinner setComponentOrientation doesn't affect on text orientation
Reviewed-by: malenkov, serb
Contributed-by: jdowland@redhat.com

Commit:
8039279: Move awt tests to openjdk repository
Reviewed-by: pchelko, alexsch
Contributed-by: Dmitriy Ermashov  <dmitriy.ermashov@oracle.com>

Commit:
8041592: [TEST_BUG] Move 42 AWT hw/lw mixing tests to jdk
Reviewed-by: anthony, pchelko

Commit:
8044172: [TEST_BUG] Move regtests for 4523758 and AltPlusNumberKeyCombinationsTest to jdk
Reviewed-by: pchelko

Commit:
8044157: [TEST_BUG] Improve recently submitted AWT_Mixing tests
Reviewed-by: pchelko

Commit:
8044765: Move functional tests AWT_SystemTray/Automated to openjdk repository
Reviewed-by: pchelko
Contributed-by: Dmitriy Ermashov  <dmitriy.ermashov@oracle.com>

Commit:
8046221: [TEST_BUG] Cleanup datatransfer tests
Reviewed-by: anthony, alexsch

Commit:
8044429: move awt automated tests for AWT_Modality to OpenJDK repository
Reviewed-by: pchelko
Contributed-by: Alexander Stepanov <alexander.v.stepanov@oracle.com>

Commit:
8047180: Move functional tests AWT_Headless/Automated to OpenJDK repository
Reviewed-by: pchelko, alexsch
Contributed-by: Dmitriy Ermashov  <dmitriy.ermashov@oracle.com>

Commit:
8043131: Move ShapedAndTranslucentWindows and GC functional AWT tests to regression tree
Reviewed-by: pchelko
Contributed-by: Dmitriy Ermashov  <dmitriy.ermashov@oracle.com>

Commit:
8048246: Move AWT_DnD/Clipboard/Automated functional tests to OpenJDK
Reviewed-by: pchelko
Contributed-by: Dmitriy Ermashov  <dmitriy.ermashov@oracle.com>

Commit:
8043126: move awt automated functional tests from AWT_Events/Lw and AWT_Events/AWT to OpenJDK repository
Reviewed-by: pchelko

Commit:
8047367: move awt automated tests from AWT_Modality to OpenJDK repository - part 2
Reviewed-by: pchelko

Commit:
8049694: Migrate functional AWT_DesktopProperties/Automated tests to OpenJDK
Reviewed-by: azvegint, serb

Commit:
8049617: move awt automated tests from AWT_Modality to OpenJDK repository - part 3
Reviewed-by: pchelko

Commit:
8051440: move tests about maximizing undecorated to OpenJDK
Reviewed-by: serb

Commit:
8050885: move awt automated tests from AWT_Modality to OpenJDK repository - part 4
Reviewed-by: pchelko

Commit:
8052408: Move AWT_BAT functional tests to OpenJDK (3 of 3)
Reviewed-by: alexsch

Commit:
8052012: move awt automated tests from AWT_Modality to OpenJDK repository - part 5
Reviewed-by: serb

Commit:
8053657: [TEST_BUG] move some 5 tests related to undecorated Frame/JFrame to JDK
Reviewed-by: serb

Commit:
8055360: Move the rest part of AWT ShapedAndTranslucent tests to OpenJDK
Reviewed-by: alexsch

Commit:
8055664: move 14 tests about setLocationRelativeTo to jdk
Reviewed-by: alexsch, azvegint

Commit:
8058805: [TEST_BUG]Test java/awt/TrayIcon/SecurityCheck/NoPermissionTest/NoPermissionTest.java fails
Reviewed-by: yan, alexsch
Contributed-by: pooja.chopra@oracle.com

Commit:
8063102: Change open awt regression tests to avoid sun.awt.SunToolkit.realSync, part 1
Reviewed-by: pchelko, serb

Commit:
8063106: Change open swing regression tests to avoid sun.awt.SunToolkit.realSync, part 1
Reviewed-by: pchelko, serb

Commit:
8055836: move awt tests from AWT_Modality to OpenJDK repository - part 9
8057694: move awt tests from AWT_Modality to OpenJDK repository - part 10
Summary: Some modality tests for Print dialog were added
Reviewed-by: pchelko

Commit:
8054359: move awt automated tests from AWT_Modality to OpenJDK repository - part 8
Summary: Some new tests to check FileDialog modality behavior
Reviewed-by: alexsch, pchelko

Commit:
8054143: move awt automated tests from AWT_Modality to OpenJDK repository - part 6
Reviewed-by: pchelko

Commit:
8063107: Change open swing regression tests to avoid sun.awt.SunToolkit.realSync, part 2
8064573: [TEST_BUG] javax/swing/text/AbstractDocument/6968363/Test6968363.java is asocial pressing VK_LEFT and not releasing
8064575: [TEST_BUG] javax/swing/JEditorPane/6917744/bug6917744.java 100 times press keys and never releases
8064809: [TEST_BUG] javax/swing/JComboBox/4199622/bug4199622.java contains a lot of keyPress and not a single keyRelease
Reviewed-by: alexsch, pchelko

Commit:
8063104: Change open awt regression tests to avoid sun.awt.SunToolkit.realSync, part 2
Reviewed-by: serb, alexsch

Commit:
8067441: Some tests fails with error: cannot find symbol getSystemMnemonicKeyCodes()
Reviewed-by: serb, azvegint

Commit:
8068275: Some tests failed after JDK-8063104
Reviewed-by: azvegint, yan

Commit:
8054358: move awt automated tests from AWT_Modality to OpenJDK repository - part 7
Reviewed-by: pchelko

Commit:
8069211: (zipfs) ZipFileSystem creates corrupted zip if entry output stream gets closed more than once
Summary: to synchronize the write and close methods of the entry output stream
Reviewed-by: alanb

Commit:
8074807: Fix some tests unnecessary using internal API
Reviewed-by: azvegint, alexsch

Commit:
8076315: move 4 manual functional swing tests to regression suite
Reviewed-by: alexsch, serb

Commit:
8130772: Util.hitMnemonics does not work: getSystemMnemonicKeyCodes() returns ALT_MASK rather than VK_ALT
Reviewed-by: serb, alexsch

Commit:
8134632: Mark javax/sound/midi/Devices/InitializationHang.java as headful
Reviewed-by: prr

Commit:
8068228: Test closed/java/awt/Mouse/MaximizedFrameTest/MaximizedFrameTest fails with GTKLookAndFeel
Reviewed-by: ssadetsky, arapte

Commit:
8148916: Mark bug6400879.java as intermittently failing
Reviewed-by: serb

Commit:
8148983: Fix extra comma in changes for JDK-8148916
Reviewed-by: jjg

Commit:
8031423: Test java/awt/dnd/DisposeFrameOnDragCrash/DisposeFrameOnDragTest.java fails by Timeout on Windows
Reviewed-by: yan, arapte

Commit:
8160438: javax/swing/plaf/nimbus/8057791/bug8057791.java fails
Reviewed-by: alexsch, rchamyal

Commit:
8163161: [PIT][TEST_BUG] increase timeout in javax/swing/plaf/nimbus/8057791/bug8057791.java
Reviewed-by: alexsch, rchamyal

Commit:
8166015: [PIT][TEST_BUG] stray character in java/awt/Focus/ModalDialogActivationTest/ModalDialogActivationTest.java
Reviewed-by: aghaisas, ssadetsky

Commit:
8132664: closed/javax/swing/DataTransfer/DefaultNoDrop/DefaultNoDrop.java locks on Windows
Reviewed-by: serb

Commit:
8168292: [TESTBUG] [macosx] Test java/awt/TrayIcon/DragEventSource/DragEventSource.java fails on OS X
Reviewed-by: psadhukhan, aghaisas

Commit:
6962725: Regtest javax/swing/JFileChooser/6738668/bug6738668.java fails under Linux
Reviewed-by: serb

Commit:
8185003: JMX: Add a version of ThreadMXBean.dumpAllThreads with a maxDepth argument
Summary: Added two new API's to limit the stack trace depth
Reviewed-by: mchung, dfuchs, rriggs, egahlin
Contributed-by: ujwal.vangapally@oracle.com

Commit:
8168682: jdk/test/java/lang/ClassLoader/forNameLeak/ClassForNameLeak.java fails with -Xcomp
Reviewed-by: bchristi

Commit:
8205507: jdk/javax/xml/crypto/dsig/GenerationTests.java timed out
Reviewed-by: mullan

Commit:
8214440: ldap over a TLS connection negotiate failed with "javax.net.ssl.SSLPeerUnverifiedException: hostname of the server '' does not match the hostname in the server's certificate"
Reviewed-by: vtewari, xuelei

Commit:
8213448: [TESTBUG] enhance jfr/jvm/TestDumpOnCrash
Summary: added 2 more crash scenarios, removed dumponexit, and more
Reviewed-by: egahlin

Commit:
8220579: [Containers] SubSystem.java out of sync with osContainer_linux.cpp
Reviewed-by: bobv, adinn

Commit:
8217766: Container Support doesn't work for some Join Controllers combinations
Reviewed-by: rriggs, sgehwolf

Commit:
8217362: Emergency dump does not work when disk=false is set
Reviewed-by: phh, andrew

Commit:
8221342: [TESTBUG] Generate Dockerfile for docker testing
Summary: Dockerfile is generated; introduced properties to specify custom base image
Reviewed-by: sgehwolf, dholmes, jiefu

Commit:
8221710: [TESTBUG] more configurable parameters for docker testing
Summary: Introduced docker test config properties
Reviewed-by: lmesnik, iignatyev, egahlin, phh

Commit:
8219013: Update Apache Santuario (XML Signature) to version 2.1.3
Reviewed-by: mullan

Commit:
8221340: [TESTBUG] TestCgroupMetrics.java fails after fix for JDK-8219562
Reviewed-by: mseledtsov, rriggs

Commit:
8220657: JFR.dump does not work when filename is set
Reviewed-by: ysuenaga

Commit:
8224502: [TESTBUG] JDK docker test TestSystemMetrics.java fails with access issues and OOM
Reviewed-by: sgehwolf, mseledtsov

Commit:
8231209: [REDO] ThreadMXBean::getThreadAllocatedBytes() can be quicker for self thread
Summary: Add com.sun.management.getCurrentThreadAllocatedBytes, implement getThreadAllocatedBytes(long) independent of getThreadAllocatedBytes(long[])
Reviewed-by: mchung, dholmes, sspitsyn

Commit:
8231968: getCurrentThreadAllocatedBytes default implementation s/b getThreadAllocatedBytes
Summary: Pass Thread.currentThread().getId() to getThreadAllocatedBytes, remove its implSpec
Reviewed-by: dholmes, mchung, sspitsyn

Commit:
8223940: Private key not supported by chosen signature algorithm
Reviewed-by: valeriep

Commit:
8234339: replace JLI_StrTok in java_md_solinux.c
Reviewed-by: clanger, rriggs

Commit:
8233548: Update CUP to v0.11b
Reviewed-by: lancea

Commit:
8223108: Test java/awt/EventQueue/NonComponentSourcePost.java is unstable
Reviewed-by: prr

Commit:
8238448: RSASSA-PSS signature verification fail when using certain odd key sizes
Summary: Calculate and set offset for correct verification for such key sizes
Reviewed-by: xuelei

Commit:
8242480: Negative value may be returned by getFreeSwapSpaceSize() in the docker
Reviewed-by: andrew

Commit:
8242335: Additional Tests for RSASSA-PSS
Summary: New Tests for RSASSA-PSS
Reviewed-by: valeriep

Commit:
8246648: issue with OperatingSystemImpl getFreeSwapSpaceSize in docker after 8242480
Reviewed-by: bobv, sgehwolf

Commit:
8247619: Improve Direct Buffering of Characters
Reviewed-by: alanb, ahgross, rhalade, psandoz, andrew

Commit:
8250665: Wrong translation for the month name of May in ar_JO,LB,SY
Reviewed-by: naoto

Commit:
8022535: [TEST BUG] javax/swing/text/html/parser/Test8017492.java fails
Reviewed-by: prr, pbansal

Commit:
8251469: Better cleanup for test/jdk/javax/imageio/SetOutput.java
Reviewed-by: prr, pbansal

Commit:
8025936: Windows .pdb and .map files does not have proper dependencies setup
Reviewed-by: andrew

Commit:
8252384: [TESTBUG] Some tests refer to COMPAT provider rather than JRE
Summary: correct backported locale tests from COMPAT to JRE locale provider
Reviewed-by: andrew
Contributed-by: jdowland@redhat.com

Commit:
8252395: [8u] --with-native-debug-symbols=external doesn't include debuginfo files for binaries
Summary: ensure debuginfo files get properly copied to the images directory
Reviewed-by: andrew, aph

Commit:
8249176: Update GlobalSignR6CA test certificates
Reviewed-by: xuelei

Commit:
8252497: Incorrect numeric currency code for ROL
Reviewed-by: naoto

Commit:
8252497: Incorrect numeric currency code for ROL
Reviewed-by: naoto

Commit:
8245400: Upgrade to LittleCMS 2.11
Reviewed-by: serb, jdv

Commit:
8051853: new URI("x/").resolve("..").getSchemeSpecificPart() returns null!
Reviewed-by: andrew

Commit:
8252975: [8u] JDK-8252395 breaks the build for --with-native-debug-symbols=internal
Reviewed-by: zgu

Commit:
8207766: [testbug] Adapt tests for Aix.
Reviewed-by: mdoerr, andrew

Commit:
Merge

Commit:
Added tag jdk8u282-b00 for changeset 4546aa3faf37

Commit:
8152545: Use preprocessor instead of compiling a program to generate native nio constants
Reviewed-by: andrew

Commit:
8253550: [8u] JDK-8252395 breaks the build for make STRIP_POLICY=no_strip
Summary: Add check that STRIP_POLICY is not no_strip before attempting to reference stripped debuginfo
Reviewed-by: aph, sgehwolf

Commit:
8252904: VM crashes when JFR is used and JFR event class is transformed
Reviewed-by: adinn

Commit:
8239105: Add exception for expiring Digicert root certificates to VerifyCACerts test
Summary: "8239105: added verisigntsaca and thawtepremiumserverca to EXPIRY_EXC_ENTRIES list"
Reviewed-by: mullan

Commit:
8254081: java/security/cert/PolicyNode/GetPolicyQualifiers.java fails due to an expired certificate
Summary: Perform backdated validation of test certificate.
Reviewed-by: mullan, xuelei

Commit:
8202076: test/jdk/java/io/File/WinSpecialFiles.java on windows with VS2017
Summary: If all other means to obtain the file length fail, fall back to FindFirstFile
Reviewed-by: igerasim, alanb, zgu

Commit:
8253752: test/sun/management/jmxremote/bootstrap/RmiBootstrapTest.java fails randomly
Reviewed-by: zgu, sgehwolf
Contributed-by: xiezhaokun@huawei.com

Commit:
8255003: Build failures on Solaris
Summary: Produce separate map file for linux
Reviewed-by: bae, andrew

Commit:
8251840: Java_sun_awt_X11_XToolkit_getDefaultScreenData should not be in make/mapfiles/libawt_xawt/mapfile-vers
Summary: Remove left-over map file entry
Reviewed-by: bae, neugens, shade
Contributed-by: Peter Tribble <peter.tribble@gmail.com>

Commit:
8254683: [TEST_BUG] jdk/test/sun/tools/jconsole/WorkerDeadlockTest.java fails
Reviewed-by: sgehwolf
Contributed-by: xiezhaokun@huawei.com

Commit:
8254177: (tz) Upgrade time-zone data to tzdata2020b
Summary: Remove obsolete systemv & pacificnew files and US/Pacific-New zone
Reviewed-by: martin

Commit:
8236512: PKCS11 Connection closed after Cipher.doFinal and NoPadding
Summary: Removed killSession() calls in certain impl classes when cancelling operations
Reviewed-by: xuelei

Commit:
8232114: JVM crashed at imjpapi.dll in native code
Reviewed-by: serb, alitvinov, andrew

Commit:
Merge

Commit:
Merge

Commit:
Merge

Commit:
8255603: Memory/Performance regression after JDK-8210985
Reviewed-by: simonis, xuelei, aph

Commit:
Added tag jdk8u282-b01 for changeset 4c5dceabd4c6

Commit:
Added tag jdk8u275-b01 for changeset efb922cd7ac4

Commit:
Added tag jdk8u275-ga for changeset efb922cd7ac4

Commit:
Merge

Commit:
Added tag jdk8u282-b02 for changeset 3d2b77f2c2fd

Commit:
8256752: 8252395 incorrect copy rule for macos .dSYM folder
Reviewed-by: sgehwolf

Commit:
Merge

Commit:
Added tag jdk8u282-b03 for changeset 23add259af02

Commit:
8229868: Update Apache Santuario TPRM version
Reviewed-by: phh

Commit:
8249846: Change of behavior after JDK-8237117: Better ForkJoinPool behavior
Reviewed-by: mbalao

Commit:
Merge

Commit:
8257397: [TESTBUG] test/lib/containers/docker/Common.java refers to -Xlog:os+container=trace
Reviewed-by: shade

Commit:
Added tag jdk8u282-b04 for changeset 78c875107d40

Commit:
8254982: (tz) Upgrade time-zone data to tzdata2020c
Reviewed-by: phh

Commit:
Added tag jdk8u282-b05 for changeset 25cd39ef5b98

Commit:
8255226: (tz) Upgrade time-zone data to tzdata2020d
Reviewed-by: sgehwolf

Commit:
Added tag jdk8u282-b06 for changeset ef358e4669db

Commit:
8225072: Add LuxTrust certificate that is expiring in March 2021 to list of allowed but expired certs
8258630: Add expiry exception for QuoVadis root certificate
Reviewed-by: ascarpino

Commit:
Added tag jdk8u282-b07 for changeset aa9f038a7034

yuleil committed:
Revert "[Backport] 8217362: Emergency dump does not work when disk=false is set"

This reverts commit 7db9d1a67398842989693ce57a1d7eb7dd206fca.

yuleil committed:
Revert "[Backport] 8213448: [TESTBUG] enhance jfr/jvm/TestDumpOnCrash"

This reverts commit 22a02da0fd075b8cf062d2ad9b4de1df6f66701e.

yuleil committed:
[Merge] Merge tag 'jdk8u282-ga'

Summary: Merge 'jdk8u282-ga' from OpenJDK

Test Plan: CI pipeline

Reviewed-by: joeyleeeeeee97, D-D-H

Issue: alibaba/dragonwell8#193

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/dragonwell8_jdk?pullRequest=48) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/dragonwell8_jdk?pullRequest=48) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: andrew<br/>:x: yuleil<br/><hr/>**andrew** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/dragonwell8_jdk?pullRequest=48) it.</sub>

yuleil committed:
[Merge] Add Dragonwell interfaces to mapfile-linux

Summary: Linux use a separate map file since 8u282,
  add Dragonwell interfaces to it.

Test Plan: build

Reviewed-by: joeyleeeeeee97, D-D-H

Issue: alibaba/dragonwell8#193

j0826 reviewed (commented):
LGTM

j0826 reviewed (approved):
LGTM

D-D-H reviewed (approved):
LGTM
